SpringProject/src/main/java/com/gradproject2019/userConference/service/UserConferenceServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.gradproject2019.userConference.service;
+
+import com.gradproject2019.auth.exception.TokenNotFoundException;
+import com.gradproject2019.auth.exception.UserUnauthorisedException;
+import com.gradproject2019.auth.service.AuthService;
+import com.gradproject2019.userConference.data.UserConferenceRequestDto;
+import com.gradproject2019.userConference.data.UserConferenceResponseDto;
+import com.gradproject2019.userConference.exception.UserAlreadyInterestedException;
+import com.gradproject2019.userConference.persistance.UserConference;
+import com.gradproject2019.userConference.repository.UserConferenceRepository;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class UserConferenceServiceImpl implements UserConferenceService {
+
+    private UserConferenceRepository userConferenceRepository;
+    private AuthService authService;
+
+    public UserConferenceServiceImpl(UserConferenceRepository userConferenceRepository, AuthService authService) {
+        this.userConferenceRepository = userConferenceRepository;
+        this.authService = authService;
+    }
+
+    @Override
+    public UserConferenceResponseDto saveInterest(UUID token, UserConferenceRequestDto userConferenceRequestDto) {
+        UserConference userConference = new UserConferenceRequestDto().from(userConferenceRequestDto);
+
+        checkUserAuthorised(token);
+
+        try {
+            UserConference savedUserConference = userConferenceRepository.saveAndFlush(userConference);
+            return new UserConferenceResponseDto().from(savedUserConference);
+        } catch (DuplicateKeyException e) {
```

Comment:
You should check whether the user is already interested before saving the conference, in a separate method

Reply:
I think that because we are using a composite key the same user cannot register interest in a conference twice, this is why we have included a catch for the DuplicateKeyException which will be thrown if the user does try to. Do we also need a method to check this if the functionality is already there?